packages/service-library/src/servicelib/aiohttp/profiler_middleware.py

```diff
@@ -1,43 +1,44 @@
+import json
+
 from aiohttp.web import HTTPInternalServerError, Request, StreamResponse, middleware
 from pyinstrument import Profiler
 from servicelib.mimetype_constants import (
     MIMETYPE_APPLICATION_JSON,
     MIMETYPE_APPLICATION_ND_JSON,
 )
 
-from .._utils_profiling_middleware import append_profile
+from .._utils_profiling_middleware import _request_profiler, append_profile
 
 
 @middleware
 async def profiling_middleware(request: Request, handler):
-    profiler: Profiler | None = None
-    if request.headers.get("x-profile") is not None:
-        profiler = Profiler(async_mode="enabled")
-        profiler.start()
-
-    response = await handler(request)
-
-    if profiler is None:
-        return response
-    if response.content_type != MIMETYPE_APPLICATION_JSON:
-        raise HTTPInternalServerError(
-            reason=f"Profiling middleware is not compatible with {response.content_type=}",
-            headers={},
+    if profiler_kwargs := request.headers.get("x-profile"):
+        profiler = Profiler(**json.loads(profiler_kwargs))
```

Comment:
minor: for info I added lately https://github.com/ITISFoundation/osparc-simcore/blob/b77814ac1ded8f8ad89ff72c43fc4434e5612a5e/packages/service-library/src/servicelib/aiohttp/requests_validation.py which might be useful.
parse_request_headers_as which takes a pydantic model.

Reply:
I decided to simplify the design a bit. Now it is not possible to enable the sync mode, but I think it is a better (less error-prone) design.